cardano-api/src/Cardano/Api/Crypto/Shelley.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE PatternSynonyms #-}
+
+module Cardano.Api.Crypto.Shelley
+  ( genKeyPair
+  ) where
+
+import           Cardano.Api.Types (KeyPair (..))
+
+import           Cardano.Crypto.DSIGN.Class (DSIGNAlgorithm (..))
+import           Cardano.Crypto.DSIGN.Ed448 ()
+
+import           Cardano.Prelude
+
+import           Crypto.Random (MonadRandom)
+
+import           Shelley.Spec.Ledger.Keys (SKey (..), pattern VKey)
+
+genKeyPair :: MonadRandom m
+           => m KeyPair
+genKeyPair = do
+  sk <- genKeyDSIGN
+  pure $ KeyPairShelley (VKey (deriveVerKeyDSIGN sk)) (SKey sk)
```

Comment:
Perhaps this should be in `cardano-ledger-specs` instead.

Reply:
Is that a separate file? If so I think its so small and trivial i should just be in `Cardano.Api`.